api/app/routers/fba.py

```diff
@@ -59,13 +73,89 @@ async def get_zones(run_type: RunType, run_datetime: datetime, for_date: date, _
+
+
+@router.get('/sfms-fuel-types/', response_model=List[SFMSFuelType])
+async def get_sfms_fuel_types():
+    """
+    Get mapping of fuel_type_id (as used by SFMS) to fuel type code/abbreviation as commonly used.
+    """
+    logger.info('/sfms-fuel-types/')
+    fuel_types = []
+    async with get_async_read_session_scope() as session:
+        rows = await get_all_sfms_fuel_types(session)
+
+        for row in rows:
+            fuel_type_object = row[0]
+            fuel_types.append(SFMSFuelType(fuel_type_id=fuel_type_object.fuel_type_id,
+                                           fuel_type_code=fuel_type_object.fuel_type_code,
+                                           description=fuel_type_object.description))
+
+        return fuel_types
+
+
+@router.get('/hfi-fuels/{run_type}/{for_date}/{run_datetime}/{zone_id}',
+            response_model=dict[int, List[ClassifiedHfiThresholdFuelTypeArea]])
+async def get_hfi_fuels_data_for_fire_zone(run_type: RunType,
+                                           for_date: date,
+                                           run_datetime: datetime,
+                                           zone_id: int):
+    """
+    Fetch rollup of fuel type/HFI threshold/area data for a specified fire zone.
+    """
+    logger.info('hfi-fuels/%s/%s/%s/%s', run_type.value, for_date, run_datetime, zone_id)
+    # get thresholds data
+    thresholds = await get_hfi_thresholds()
+    # get fuel type ids data
+    fuel_types = await get_sfms_fuel_types()
+
+    async with get_async_read_session_scope() as session:
+
+        # get HFI/fuels data for specific zone
+        hfi_fuel_type_ids_for_zone = await get_high_hfi_fuel_types_for_zone(session,
+                                                                            run_type=RunTypeEnum(run_type.value),
+                                                                            for_date=for_date,
+                                                                            run_datetime=run_datetime,
+                                                                            zone_id=zone_id)
+        data = []
+
+        for record in hfi_fuel_type_ids_for_zone:
+            fuel_type_id = record[1]
+            threshold_id = record[2]
+            # area is stored in square metres in DB. For user convenience, convert to hectares
+            # 1 ha = 10,000 sq.m.
+            area = record[3] / 10000
+            fuel_type_obj = next((ft for ft in fuel_types if ft.fuel_type_id == fuel_type_id), None)
+            threshold_obj = next((th for th in thresholds if th.id == threshold_id), None)
+            data.append(ClassifiedHfiThresholdFuelTypeArea(fuel_type=fuel_type_obj, threshold=threshold_obj, area=area))
+
+        return {zone_id: data}
+
+
 @router.get('/hfi-fuels/{run_type}/{for_date}/{run_datetime}',
             response_model=dict[str, List[HfiThresholdAreaByFuelType]])
 async def get_hfi_thresholds_by_fuel_type(run_type: RunType,
```

Comment:
Looks like we can delete this now since we only call the implementation with the zone parameter